SWENT-GRP5/SOLVIT
# Service Requests  & Overview

### Description:
This pull request refines the RequestsOverview functionality by displaying only the requests created by the logged-in user, enhancing the personalized experience for service seekers. Additionally, it associates each request with the userId of the logged-in seeker upon creation, ensuring accurate tracking and data integrity.

- **What issue does it solve?** This PR improves user experience by displaying only relevant requests to each user, aligning with personalized content delivery.
- **What functionality does it add?** It updates the logic in RequestsOverviewScreen and VM to filter requests by user, allowing users to view and modify only their own service requests. It also associates each new request with the correct userId to facilitate better data management.

### Related Issues:
#246 and #245 

### Changes Made:
- Modified `RequestsOverviewScreen` to display only requests belonging to the logged-in user.
- Modified `CreateRequestScreen` to attach the logged user ID to the request upon creation.
- Updated `ServiceRequest` and `ServiceRequestRepositoryFirebase` to adapt the document decoding after changing one of the requests' field by "userId".
- Adapted unit tests to align with recent modifications in the service requests.
- Applied ktfmt formatting to ensure consistent code style.

### How to Test:
1. Log in as a specific user and verify that only the requests created by that user are visible in the RequestsOverview.
2. Create a new request and confirm that the request is associated with the logged-in user’s userId.
3. Run the updated unit tests and AndroidTests to validate the functionality of the remaining tests.

---

### Checklist:
